src/test-apps/happy/test-templates/WeaveInetDNS.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+
+
+#
+#    Copyright (c) 2016-2017 Nest Labs, Inc.
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+#
+#    @file
+#       Implements WeaveInet class that tests Weave Inet Layer among Weave Nodes.
+#
+
+import os
+import sys
+import time
+
+from happy.ReturnMsg import ReturnMsg
+from happy.Utils import *
+from happy.HappyNode import HappyNode
+from happy.HappyNetwork import HappyNetwork
+
+from WeaveTest import WeaveTest
+
+
+# Q: what are the parameters need to specify?
+options = {}
+options["quiet"] = False
+options["node_id"] = None
+options["tap_if"] = None
+options["node_ip"] = None
+options["ipv4_gateway"] = None
+options["dns"] = None
+options["use_lwip"] = False
+
+
+def option():
+    return options.copy()
+
+
+class WeaveInetDNS(HappyNode, HappyNetwork, WeaveTest):
+    
+    def __init__(self, opts = options):
+        HappyNode.__init__(self)
+        HappyNetwork.__init__(self)
+        WeaveTest.__init__(self)
+
+        self.quiet = opts["quiet"]
+        self.node_id = opts["node_id"]
+        self.tap_if = options["tap_if"]
+        self.node_ip = options["node_ip"]
```

Comment:
See prior comments. We should to derive this from a prefix or just extract this from the topology.

Reply:
??I will check with you for this one